PROTO_tests/tests/dtypes_test.py

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+from arkouda import dtypes
+
+"""
+DtypesTest encapsulates arkouda dtypes module methods
+"""
+
+SUPPORTED_NP_DTYPES = [
+    bool,
+    int,
+    float,
+    str,
+    np.bool_,
+    np.int64,
+    np.float64,
+    np.uint8,
+    np.uint64,
+    np.str_,
+]
+
+
+class TestDTypes:
+    @pytest.mark.parametrize("dtype", SUPPORTED_NP_DTYPES)
+    def test_check_np_dtype(self, dtype):
+        dtypes.check_np_dtype(np.dtype(dtype))
+
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype(np.dtype(np.int16))
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype("np.str")
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype("bigint")
+
+    def test_translate_np_dtype(self):
+        for b in [np.bool_, bool]:
+            assert ("bool", 1) == dtypes.translate_np_dtype(np.dtype(b))
+
+        for s in [np.str_, str]:
+            assert ("str", 0) == dtypes.translate_np_dtype(np.dtype(s))
+
+        assert ("int", 8) == dtypes.translate_np_dtype(np.dtype(np.int64))
+        assert ("uint", 8) == dtypes.translate_np_dtype(np.dtype(np.uint64))
+        assert ("float", 8) == dtypes.translate_np_dtype(np.dtype(np.float64))
+        assert ("uint", 1) == dtypes.translate_np_dtype(np.dtype(np.uint8))
+
+    @pytest.mark.parametrize(
+        "dtype, val",
+        [
+            ("bool", True),
+            ("int64", 1),
+            ("float64", float(0.0)),
+            ("str", "test"),
+            ("int64", np.int64(1)),
+            ("<class 'list'>", [1]),
+            ("float64", np.nan),
+            ("float64", -np.inf),
+        ],
+    )
+    def test_resolve_scalar_dtype(self, dtype, val):
+        assert dtype == dtypes.resolve_scalar_dtype(val)
```

Comment:
I think most of these parameterizations should be refactored. In particular the cases when it's used only once and declared in the `pytest.mark.parametrize` block

Reply:
Done